tsdb/engine/tsm1/engine.go

```diff
@@ -1352,7 +1360,7 @@ func (e *Engine) createVarRefIterator(measurement string, opt influxql.IteratorO
 	}
 
 	// Determine tagsets for this measurement based on dimensions and filters.
-	tagSets, err := mm.TagSets(e.id, opt.Dimensions, opt.Condition)
+	tagSets, err := mm.TagSets(e.id, opt.Dimensions, opt.Condition, opt)
```

Comment:
Should this just take a `<-chan struct{}` instead? If we want to change it to take the iterator options, we should just remove the `opt.Dimensions` and `opt.Condition` and pass that as part of the IteratorOptions.